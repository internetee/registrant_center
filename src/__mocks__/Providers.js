import { CookiesProvider } from 'react-cookie';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import translations from '../translations';
import state from './mockReduxState';

const lang = 'et';

export default function Providers({ children }) {
    return (
        <BrowserRouter>
            <Provider store={state}>
                <CookiesProvider>
                    <IntlProvider
                        key={lang}
                        defaultLocale="et"
                        locale={lang}
                        messages={translations[lang]}
                    >
                        {children}
                    </IntlProvider>
                </CookiesProvider>
            </Provider>
        </BrowserRouter>
    );
}
