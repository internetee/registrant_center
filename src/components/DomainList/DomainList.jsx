import React, { useEffect, useRef, useState } from 'react';
import MediaQuery from 'react-responsive';
import { FormattedMessage, useIntl } from 'react-intl';
import {
    Table,
    Button,
    Form,
    Input,
    Icon,
    Transition,
    Dropdown,
    Container,
    Pagination,
} from 'semantic-ui-react';
import Masonry from 'react-masonry-component';
import Flatpickr from 'react-flatpickr';
import { Estonian } from 'flatpickr/dist/l10n/et';
import 'flatpickr/dist/themes/light.css';
import classNames from 'classnames';
import { useCookies } from 'react-cookie';
import DomainGridItem from './GridItem';
import DomainListItem from './ListItem';
import PageMessage from '../PageMessage/PageMessage';
import domainStatuses from '../../utils/domainStatuses.json';

const masonryOptions = {
    columnWidth: '.domains-grid--item',
    gutter: 0,
    horizontalOrder: true,
    initLayout: true,
    itemSelector: '.domains-grid--item',
    percentPosition: true,
    transitionDuration: 300,
};

const perPageOptions = [
    { key: 6, text: '6', value: 6 },
    { key: 12, text: '12', value: 12 },
    { key: 24, text: '24', value: 24 },
];

const getLocale = (locale) => {
    if (locale === 'et') {
        return Estonian;
    }
    return {};
};

const DomainList = ({ domainCount, domainTotal, domains, lang, checked, isTech }) => {
    const { formatMessage } = useIntl();
    const [cookies, setCookies] = useCookies(['domainsIsGrid']);
    const { domainsIsGrid, domainsPerPage } = cookies;
    const [isGrid, setIsGrid] = useState(domainsIsGrid ? JSON.parse(domainsIsGrid) : true);
    const [isAdvSearchOpen, setIsAdvSearchOpen] = useState(false);
    const [form, setForm] = useState({
        queryKeys: '',
        queryRegistrant: 'all',
        queryRole: isTech,
        queryStatus: 'all',
        queryValidToMax: null,
        queryValidToMin: null,
    });
    const {
        queryKeys,
        queryRegistrant,
        queryStatus,
        queryValidToMin,
        queryValidToMax,
        queryRole 
    } = form;
    const [minValidToDate, setMinValidToDate] = useState(null);
    const [maxValidToDate, setMaxValidToDate] = useState(null);
    const [filteredDomains, setFilteredDomains] = useState([]);
    const [registrantsOptions, setRegistrantsOptions] = useState([]);
    const [statuses, setStatuses] = useState([]);
    const [activePage, setActivePage] = useState(1);
    const [perPage, setPerPage] = useState(domainsPerPage ? Number(domainsPerPage) : 24);
    const [domainsList, setDomainsList] = useState([]);
    const totalDomains = domainCount || 0;

    const masonry = useRef(null);

    useEffect(() => {
        if (domains.length) {
            const { registrants, allStatuses, maxDate, minDate, sortedDomains } = domains.reduce(
                (acc, domain) => ({
                    ...acc,
                    maxDate: domain.valid_to > acc.maxDate ? domain.valid_to : acc.maxDate,
                    minDate: domain.valid_to < acc.minDate ? domain.valid_to : acc.minDate,
                    registrants: {
                        ...acc.registrants,
                        [domain.registrant.ident]: {
                            text: domain.registrant.name,
                            value: domain.registrant.ident,
                        },
                    },
                    sortedDomains: [
                        ...acc.sortedDomains,
                        {
                            ...domain,
                            statuses: domain.statuses.sort(
                                (a, b) => domainStatuses[a].priority - domainStatuses[b].priority
                            ),
                        },
                    ],
                    statuses: {
                        ...acc.statuses,
                        ...domain.statuses.reduce(
                            (statusesAcc, status) => ({
                                ...statusesAcc,
                                [status]: {
                                    key: status,
                                    label: {
                                        circular: true,
                                        color: domainStatuses[status]?.color || 'grey',
                                        empty: true,
                                    },
                                    text: formatMessage({ id: `domain.status.${status}` }),
                                    value: status,
                                },
                            }),
                            {}
                        ),
                    },
                }),
                {
                    allStatuses: {
                        all: {
                            key: 0,
                            label: {
                                circular: true,
                                color: domainStatuses.all.color,
                                empty: true,
                            },
                            text: formatMessage({ id: 'domain.status.all' }),
                            value: 'all',
                        },
                    },
                    maxDate: domains[0].valid_to,
                    minDate: domains[0].valid_to,
                    registrants: {
                        all: {
                            key: 0,
                            text: formatMessage({ id: 'domainsList.allRegistrants' }),
                            value: 'all',
                        },
                    },
                    sortedDomains: [],
                }
            );

            setStatuses(
                Object.values(allStatuses).sort(
                    (a, b) => domainStatuses[a].priority - domainStatuses[b].priority
                )
            );

            const reg_list = Object.values(registrants);
            reg_list.some(
                (i, idx) => i.value == 'all' && reg_list.unshift(reg_list.splice(idx, 1)[0])
            );

            setRegistrantsOptions(reg_list);
            setFilteredDomains(sortedDomains);
            setDomainsList(sortedDomains);
            setMinValidToDate(new Date(minDate));
            setMaxValidToDate(new Date(maxDate));
        }
    }, [domains, formatMessage, lang]);

    // Set valid_to date range datepicker options
    const dateStartOptions = {
        dateFormat: 'd.m.Y',
        defaultDate: minValidToDate,
        locale: getLocale(lang),
        maxDate: maxValidToDate,
        minDate: minValidToDate,
    };

    const dateEndOptions = {
        dateFormat: 'd.m.Y',
        defaultDate: maxValidToDate,
        locale: getLocale(lang),
        maxDate: maxValidToDate,
        minDate: minValidToDate,
    };

    const paginatedDomains = [];
    const copied = [...filteredDomains];
    const numOfChild = Math.ceil(copied.length / perPage);
    for (let i = 0; i < numOfChild; i += 1) {
        paginatedDomains.push(copied.splice(0, perPage));
    }

    const toggleView = () => {
        setCookies('domainsIsGrid', !isGrid, { path: '/' });
        setActivePage(1);
        setIsGrid((prevState) => !prevState);
        setTimeout(() => {
            if (filteredDomains.length > 0) {
                masonry.current.performLayout();
            }
        });
    };

    const toggleAdvSearch = () => {
        setIsAdvSearchOpen((prevState) => !prevState);
    };

    const setMinDate = (selectedDates) => {
        setForm((prevState) => ({
            ...prevState,
            queryValidToMin: new Date(selectedDates[0]).getTime(),
        }));
    };

    const setMaxDate = (selectedDates) => {
        setForm((prevState) => ({
            ...prevState,
            queryValidToMax: new Date(selectedDates[0]).setHours(23, 59, 59),
        }));
    };

    const handleItemsPerPage = (event, { value }) => {
        setCookies('domainsPerPage', value, { path: '/' });
        setActivePage(1);
        setPerPage(value);
    };

    const handleChange = (event, { name, value }) => {
        setForm((prevState) => ({
            ...prevState,
            [name]: value,
        }));
    };

    const handleSubmit = () => {
        if (
            !queryKeys.length &&
            queryStatus === 'all' &&
            !queryValidToMin &&
            !queryValidToMax &&
            queryRegistrant === 'all'
        ) {
            setActivePage(1);
            setFilteredDomains(domainsList);
        } else {
            setActivePage(1);
            setFilteredDomains(
                domainsList
                    .filter((domain) => {
                        const query = queryKeys.toString().toLowerCase();
                        if (query.length) {
                            return domain.name.toLowerCase().includes(query);
                        }
                        return true;
                    })
                    .filter((domain) => {
                        if (queryStatus.length && queryStatus !== 'all') {
                            return domain.statuses.includes(queryStatus);
                        }
                        return true;
                    })
                    .filter((domain) => {
                        if (queryValidToMin && queryValidToMax) {
                            return (
                                new Date(domain.valid_to).getTime() >= queryValidToMin &&
                                new Date(domain.valid_to).getTime() <= queryValidToMax
                            );
                        }
                        if (queryValidToMin && !queryValidToMax) {
                            return new Date(domain.valid_to).getTime() >= queryValidToMin;
                        }
                        if (!queryValidToMin && queryValidToMax) {
                            return new Date(domain.valid_to).getTime() <= queryValidToMax;
                        }
                        return true;
                    })
                    .filter((domain) => {
                        if (queryRegistrant.length && queryRegistrant !== 'all') {
                            return domain.registrant.ident.includes(queryRegistrant);
                        }
                        return true;
                    })
            );
        }
    };

    const handleReset = () => {
        setForm({
            queryKeys: '',
            queryRegistrant: 'all',
            queryStatus: 'all',
            queryValidToMax: null,
            queryValidToMin: null,
        });
        setFilteredDomains(domainsList);
    };

    const roleOptions = [
        {
            id: 'domains.roles.regAndAdmRoles',
            key: 'domains.roles.regAndAdmRoles',
            text: formatMessage({ id: 'domains.roles.regAndAdmRoles' }),
            value: 'domains.roles.regAndAdmRoles',
        },
        {
            id: 'domains.roles.allRoles',
            key: 'domains.roles.allRoles',
            text: formatMessage({ id: 'domains.roles.allRoles' }),
            value: 'domains.roles.allRoles',
        },
    ];

    const handleRole = (event, { name, value }) => {
        if (value === 'domains.roles.regAndAdmRoles' && isTech) 
        {
            checked(false);
        }
        if (value === 'domains.roles.allRoles' && !isTech) 
        {
            checked(true);
        }
    };

    const setRoleValue = () => {
        if (isTech) return 'domains.roles.allRoles';
        return 'domains.roles.regAndAdmRoles';
    };

    return (
        <div className="domains-list--wrap">
            <div className="page--header">
                <Container>
                    <div className="page--header--text">
                        <FormattedMessage
                            id="domains.title"
                            tagName="h2"
                            values={{
                                span: (text) => <span>{text} of {domainTotal}</span>,
                                userTotalDomains: totalDomains,
                            }}
                        />
                    </div>
                    <Form className="form-filter" onSubmit={handleSubmit}>
                        <div className="form-filter--search">
                            <div className="form-filter--actions" />
                            <div className="search-field">
                                <Input
                                    className="icon"
                                    name="queryKeys"
                                    onChange={handleChange}
                                    placeholder={formatMessage({
                                        id: 'domains.searchForADomain',
                                    })}
                                    size="massive"
                                    type="text"
                                    value={queryKeys}
                                />
                                <Button
                                    color="orange"
                                    icon="sync"
                                    onClick={handleReset}
                                    type="reset"
                                />
                                <Button icon="arrow right" primary type="submit" />
                            </div>
                            <div className="form-filter--actions">
                                <Icon
                                    className={classNames('action--filter', {
                                        active: isAdvSearchOpen,
                                    })}
                                    link
                                    name="filter"
                                    onClick={toggleAdvSearch}
                                    size="big"
                                />
                                <MediaQuery query="(min-width: 768px)" values={{ width: 1224 }}>
                                    <Icon
                                        className={classNames('action--list', { active: !isGrid })}
                                        link
                                        name="th list"
                                        onClick={toggleView}
                                        size="big"
                                    />
                                    <Icon
                                        className={classNames('action--grid', { active: isGrid })}
                                        link
                                        name="th"
                                        onClick={toggleView}
                                        size="big"
                                    />
                                </MediaQuery>
                            </div>
                        </div>
                        <Transition animation="slide down" visible={isAdvSearchOpen}>
                            <div className="form-filter--adv-search">
                                <span style={{  display: "flex",
                                                justifyContent: "center", 
                                                margin: "10px", 
                                                textAlign: "center"
                                            }} 
                                />
                                <Form.Group>
                                    <Form.Field width="5">
                                        <FormattedMessage
                                            id="domains.form.selectRegistrant"
                                            tagName="label"
                                        />
                                        <Dropdown
                                            fluid
                                            name="queryRegistrant"
                                            onChange={handleChange}
                                            options={registrantsOptions}
                                            search
                                            selection
                                            value={queryRegistrant}
                                        />
                                    </Form.Field>
                                    <Form.Field width="5">
                                        <FormattedMessage
                                            id="domains.form.dateRange"
                                            tagName="label"
                                        />
                                        <Form.Group className="date-range">
                                            <div className="field">
                                                <div className="ui right icon input">
                                                    <Flatpickr
                                                        onClose={setMinDate}
                                                        options={dateStartOptions}
                                                    />
                                                    <Icon link name="calendar alternate" />
                                                </div>
                                            </div>
                                            <span className="date-range-sep">–</span>
                                            <div className="field">
                                                <div className="ui right icon input">
                                                    <Flatpickr
                                                        onClose={setMaxDate}
                                                        options={dateEndOptions}
                                                    />
                                                    <Icon link name="calendar alternate" />
                                                </div>
                                            </div>
                                        </Form.Group>
                                    </Form.Field>
                                    <Form.Field width="6">
                                        <FormattedMessage
                                            id="domains.form.selectStatus"
                                            tagName="label"
                                        />
                                        <Dropdown
                                            fluid
                                            name="queryStatus"
                                            onChange={handleChange}
                                            options={statuses}
                                            selection
                                            value={queryStatus}
                                        />
                                    </Form.Field>
                                    <Form.Field width="4">
                                    <FormattedMessage
                                            id="domains.form.selectRole"
                                            tagName="label"
                                        />
                                        <Dropdown
                                            defaultValue=
                                            { 
                                                isTech ? roleOptions[1].value : roleOptions[0].value 
                                            }
                                            fluid
                                            name="queryRole"
                                            onChange={handleRole}
                                            options={roleOptions}
                                            selection
                                        />    
                                    </Form.Field>
                                    <div className="form-actions">
                                        <Button primary>
                                            <FormattedMessage
                                                id="domains.form.filter"
                                                tagName="span"
                                            />
                                        </Button>
                                    </div>
                                </Form.Group>
                            </div>
                        </Transition>
                    </Form>
                </Container>
            </div>
            {domains.length ? (
                <>
                    {paginatedDomains.length ? (
                        <>
                            <Transition animation="fade" duration={300} visible={isGrid}>
                                <div className="domains-grid--wrap">
                                    <Masonry
                                        ref={masonry}
                                        className="domains-grid"
                                        disableImagesLoaded
                                        enableResizableChildren
                                        options={masonryOptions}
                                    >
                                        {paginatedDomains[activePage - 1].map((domain) => (
                                            <DomainGridItem
                                                key={domain.id}
                                                domain={domain}
                                                lang={lang}
                                            />
                                        ))}
                                    </Masonry>
                                </div>
                            </Transition>
                            <Transition animation="fade" duration={300} visible={!isGrid}>
                                <div>
                                    <div className="domains-list">
                                        <Table celled unstackable verticalAlign="top">
                                            <Table.Header>
                                                <Table.Row>
                                                    <Table.HeaderCell>
                                                        <FormattedMessage
                                                            id="domains.domainName"
                                                            tagName="span"
                                                        />
                                                    </Table.HeaderCell>
                                                    <Table.HeaderCell>
                                                        <FormattedMessage
                                                            id="domains.registrar"
                                                            tagName="span"
                                                        />
                                                    </Table.HeaderCell>
                                                    <Table.HeaderCell>
                                                        <FormattedMessage
                                                            id="domains.status"
                                                            tagName="span"
                                                        />
                                                    </Table.HeaderCell>
                                                    <Table.HeaderCell>
                                                        <FormattedMessage
                                                            id="domains.validUntil"
                                                            tagName="span"
                                                        />
                                                    </Table.HeaderCell>
                                                </Table.Row>
                                            </Table.Header>
                                            <Table.Body>
                                                {paginatedDomains[activePage - 1].map((domain) => (
                                                    <DomainListItem
                                                        key={domain.id}
                                                        domain={domain}
                                                        lang={lang}
                                                    />
                                                ))}
                                            </Table.Body>
                                        </Table>
                                    </div>
                                </div>
                            </Transition>
                            <div className="paginator">
                                <Container>
                                    <Pagination
                                        activePage={activePage}
                                        firstItem={null}
                                        lastItem={null}
                                        nextItem={{
                                            content: (
                                                <>
                                                    <FormattedMessage
                                                        id="pagination.next"
                                                        tagName="span"
                                                    />
                                                    <Icon name="arrow right" />
                                                </>
                                            ),
                                            disabled: activePage === paginatedDomains.length,
                                            icon: true,
                                        }}
                                        onPageChange={(e, { activePage: page }) =>
                                            setActivePage(page)
                                        }
                                        prevItem={{
                                            content: (
                                                <>
                                                    <Icon name="arrow left" />
                                                    <FormattedMessage
                                                        id="pagination.previous"
                                                        tagName="span"
                                                    />
                                                </>
                                            ),
                                            disabled: activePage === 1,
                                            icon: true,
                                        }}
                                        totalPages={paginatedDomains.length}
                                    />
                                    <Form>
                                        <Form.Field inline>
                                            <FormattedMessage
                                                id="pagination.perPage"
                                                tagName="label"
                                            />
                                            <Dropdown
                                                onChange={handleItemsPerPage}
                                                options={perPageOptions}
                                                selection
                                                value={perPage}
                                            />
                                        </Form.Field>
                                    </Form>
                                </Container>
                            </div>
                        </>
                    ) : (
                        <PageMessage
                            headerContent={<FormattedMessage id="domains.search.message.title" />}
                        />
                    )}
                </>
            ) : (
                <PageMessage
                    headerContent={
                        <FormattedMessage id="domains.none.message.title" tagName="span" />
                    }
                    icon="frown outline"
                />
            )}
        </div>
    );
};

export default DomainList;
