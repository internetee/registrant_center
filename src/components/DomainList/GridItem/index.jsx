import React, { useState } from 'react';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import { FormattedMessage, useIntl } from 'react-intl';
import { Icon, Label, Table, Transition } from 'semantic-ui-react';
import moment from 'moment';
import 'moment/locale/et';
import 'moment/locale/ru';
import domainStatuses from '../../../utils/domainStatuses.json';

const DomainGridItem = ({ domain, lang }) => {
    moment.locale(lang);
    const { formatMessage } = useIntl();
    const [isOpen, setIsOpen] = useState(false);
    const [showStatuses, setShowStatuses] = useState(false);

    const handleExtra = () => {
        setIsOpen((prevState) => !prevState);
        setShowStatuses((prevState) => !prevState);
    };

    const handleStatuses = () => {
        setShowStatuses((prevState) => !prevState);
    };

    if (domain) {
        return (
            <article className={classNames('domains-grid--item', { 'is-open': isOpen })}>
                <div className="container">
                    <div className="content">
                        <Link className="link" to={`/domain/${domain.id}`}>
                            <h2>{domain.name}</h2>
                        </Link>
                        <p>
                            {domain.registrar && domain.registrar.website ? (
                                <a
                                    href={
                                        domain.registrar.website.indexOf('http') > -1
                                            ? domain.registrar.website
                                            : `http://${domain.registrar.website}`
                                    }
                                    rel="noopener noreferrer"
                                    target="_blank"
                                >
                                    {domain.registrar.name}
                                </a>
                            ) : (
                                '-'
                            )}
                        </p>
                        <Label.Group className="statuses" size="large">
                            {domain.statuses.map((status, i) => (
                                <Label
                                    key={status}
                                    basic
                                    circular
                                    className={classNames({ hidden: !showStatuses && i > 0 })}
                                    color={domainStatuses[status].color}
                                    title={formatMessage({
                                        id: `domain.status.${status}.description`,
                                    })}
                                >
                                    <Icon name="circle" />
                                    <FormattedMessage
                                        id={`domain.status.${status}`}
                                        tagName="span"
                                    />
                                </Label>
                            ))}
                            <Label
                                as="a"
                                circular
                                className={classNames({ hidden: domain.statuses.length < 2 })}
                                color="black"
                                onClick={handleStatuses}
                                role="button"
                            >
                                {domain.statuses.length > 1 &&
                                    (showStatuses
                                        ? `-${domain.statuses.length - 1}`
                                        : `+${domain.statuses.length - 1}`)}
                            </Label>
                        </Label.Group>
{/*                        <Transition
                            animation={isOpen ? 'slide down' : 'slide down'}
                            duration={300}
                            unmountOnHide
                            visible={isOpen}
                        >
                            <div className="extra">
                                <h5>
                                    <FormattedMessage id="domains.domain.contacts.title" />
                                </h5>
                                <div className="table-wrap">
                                    <Table basic="very" compact size="small" unstackable>
                                        <Table.Header>
                                            <Table.Row>
                                                <Table.HeaderCell>
                                                    <FormattedMessage
                                                        id="domains.domain.contacts.type"
                                                        tagName="strong"
                                                    />
                                                </Table.HeaderCell>
                                                <Table.HeaderCell>
                                                    <FormattedMessage
                                                        id="domains.domain.contacts.name"
                                                        tagName="strong"
                                                    />
                                                </Table.HeaderCell>
                                                <Table.HeaderCell>
                                                    <FormattedMessage
                                                        id="domains.domain.contacts.email"
                                                        tagName="strong"
                                                    />
                                                </Table.HeaderCell>
                                            </Table.Row>
                                        </Table.Header>
                                        <Table.Body>
                                            <DomainContacts
                                                contacts={domain.tech_contacts}
                                                type="tech"
                                            />
                                            <DomainContacts
                                                contacts={domain.admin_contacts}
                                                type="admin"
                                            />
                                        </Table.Body>
                                    </Table>
                                </div>
                                <h5>
                                    <FormattedMessage id="domains.domain.nameservers.title" />
                                </h5>
                                {domain.nameservers.length ? (
                                    <div className="table-wrap">
                                        <Table basic="very" compact size="small" unstackable>
                                            <Table.Header>
                                                <Table.Row>
                                                    <Table.HeaderCell>
                                                        <FormattedMessage
                                                            id="domains.domain.nameservers.hostname"
                                                            tagName="strong"
                                                        />
                                                    </Table.HeaderCell>
                                                    <Table.HeaderCell>
                                                        <strong>IPv4</strong>
                                                    </Table.HeaderCell>
                                                    <Table.HeaderCell>
                                                        <strong>IPv6</strong>
                                                    </Table.HeaderCell>
                                                </Table.Row>
                                            </Table.Header>
                                            <Table.Body>
                                                <DomainNameservers
                                                    nameservers={domain.nameservers}
                                                />
                                            </Table.Body>
                                        </Table>
                                    </div>
                                ) : (
                                    <FormattedMessage id="domain.nameservers.text" tagName="p" />
                                )}
                            </div>
                        </Transition>*/}
                        <div className="data">
                            {domain.valid_to && (
                                <FormattedMessage
                                    id="domains.domain.validTo"
                                    tagName="p"
                                    values={{
                                        strong: (text) => <strong>{text}</strong>,
                                        valid_to: moment(domain.valid_to).format('DD.MM.Y HH:mm'),
                                    }}
                                />
                            )}
                            {domain.outzone_at && (
                                <FormattedMessage
                                    id="domains.domain.outzoneAt"
                                    tagName="p"
                                    values={{
                                        outzone_at: moment(domain.outzone_at).format(
                                            'DD.MM.Y HH:mm'
                                        ),
                                        strong: (text) => <strong>{text}</strong>,
                                    }}
                                />
                            )}
                            {domain.delete_at && (
                                <FormattedMessage
                                    id="domains.domain.deleteAt"
                                    tagName="p"
                                    values={{
                                        delete_at: moment(domain.delete_at).format('DD.MM.Y HH:mm'),
                                        strong: (text) => <strong>{text}</strong>,
                                    }}
                                />
                            )}
                            {domain.registrant_verification_asked_at && (
                                <FormattedMessage
                                    id="domains.domain.registrantVerificationAskedAt"
                                    tagName="p"
                                    values={{
                                        registrant_verification_asked_at: moment(
                                            domain.registrant_verification_asked_at
                                        ).format('DD.MM.Y HH:mm'),
                                        strong: (text) => <strong>{text}</strong>,
                                    }}
                                />
                            )}
                        </div>
                    </div>
                    {/*<button className="toggle" onClick={handleExtra} type="button">
                        <Icon name={isOpen ? 'caret up' : 'caret down'} />
                    </button>*/}
                </div>
            </article>
        );
    }
    return null;
};

const DomainNameservers = ({ nameservers }) => {
    return nameservers.map((item) => (
        <Table.Row key={item.ipv4}>
            <Table.Cell width="5">{item.hostname}</Table.Cell>
            <Table.Cell>{item.ipv4}</Table.Cell>
            <Table.Cell>{item.ipv6 ? item.ipv6 : '–'}</Table.Cell>
        </Table.Row>
    ));
};

const DomainContacts = ({ contacts, type }) =>
    contacts.map((item) => (
        <Table.Row key={item.id}>
            <Table.Cell width="3">
                {type === 'admin' && (
                    <FormattedMessage
                        defaultMessage="Admin"
                        id="domains.domain.contacts.admin"
                        tagName="strong"
                    />
                )}
                {type === 'tech' && (
                    <FormattedMessage
                        defaultMessage="Tehniline"
                        id="domains.domain.contacts.tech"
                        tagName="strong"
                    />
                )}
            </Table.Cell>
            <Table.Cell>{item.name}</Table.Cell>
            <Table.Cell>
                {item.email && <a href={`mailto:${item.email}`}>{item.email}</a>}
            </Table.Cell>
        </Table.Row>
    ));

export default DomainGridItem;
