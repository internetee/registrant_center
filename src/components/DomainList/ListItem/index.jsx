import React, { useState } from 'react';
import { Icon, Label, Table } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import moment from 'moment';

import PropTypes from 'prop-types';

import 'moment/locale/et';
import 'moment/locale/ru';
import { FormattedMessage, useIntl } from 'react-intl';
import domainStatuses from '../../../utils/domainStatuses.json';

const DomainListItem = ({ domain, lang }) => {
    moment.locale(lang);
    const { formatMessage } = useIntl();
    const [showStatuses, setShowStatuses] = useState(false);
    const handleStatuses = () => {
        setShowStatuses((prevState) => !prevState);
    };
    return (
        <Table.Row key={domain.id}>
            <Table.Cell>
                <Link to={`/domain/${domain.id}`}>{domain.name}</Link>
            </Table.Cell>
            <Table.Cell>
                <a
                    href={
                        domain.registrar.website.indexOf('http') > -1
                            ? domain.registrar.website
                            : `//${domain.registrar.website}`
                    }
                    rel="noopener noreferrer"
                    target="_blank"
                >
                    {domain.registrar.name}
                </a>
            </Table.Cell>
            <Table.Cell width={4}>
                <Label.Group className="statuses" size="large">
                    {domain.statuses.map((status, i) => (
                        <Label
                            key={status}
                            basic
                            circular
                            className={classNames({ hidden: !showStatuses && i > 0 })}
                            color={domainStatuses[status].color}
                            title={formatMessage({
                                id: `domain.status.${status}.description`,
                            })}
                        >
                            <Icon name="circle" />
                            <FormattedMessage id={`domain.status.${status}`} tagName="span" />
                        </Label>
                    ))}
                    <Label
                        as="a"
                        circular
                        className={classNames({ hidden: domain.statuses.length < 2 })}
                        color="black"
                        onClick={handleStatuses}
                        role="button"
                    >
                        {domain.statuses.length > 1 &&
                            (showStatuses
                                ? `-${domain.statuses.length - 1}`
                                : `+${domain.statuses.length - 1}`)}
                    </Label>
                </Label.Group>
            </Table.Cell>
            <Table.Cell width={3}>{moment(domain.valid_to).format('DD.MM.Y HH:mm')}</Table.Cell>
        </Table.Row>
    );
};

export default DomainListItem;

DomainListItem.propTypes = {
    domain: PropTypes.object.isRequired,
    lang: PropTypes.string.isRequired,
};
