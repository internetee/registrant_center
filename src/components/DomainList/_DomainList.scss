.dialog-company-contacts-box {
  width: 600px;
  display: flex;
  flex-direction: column;
  justify-content: center;

  border: 1px dashed red;
  background-color: rgb(216, 121, 121);
  padding: 4px;
  margin: 0 auto;

  button {
    width: 200px;
    margin: 20px auto !important;
    // margin-top: 20px !important;
    background-color: rgb(107, 206, 148) !important;
  }
}

.domains-list--wrap {
  .paginator {
    margin: 0 0 40px;
  }
  .ui.form.form-filter .form-filter--search {
    .search-field {
      .input input {
        padding-right: 7.143em !important;
        @include min-screen(768px) {
          padding-right: 6.429em !important;
        }
      }
    }
  }
}

.domains-grid {
  position: relative;
  z-index: 2;
  width: 1380px;
  max-width: 100%;
  margin: 0 auto;
  &--item {
    width: 100%;
    margin-bottom: 20px;
    text-align: center;
    &.is-open {
      .toggle {
        color: #fff;
        background: #A7A8AA;
      }
    }
    .link {
      display: block;
      padding: 20px;
    }
    .container {
      display: flex;
      min-height: 100%;
      color: #212224;
      background: #fff;
      box-shadow: 0 0 20px rgba(0,0,0,.1);
    }
    .content {
      flex: 1 1 100%;
      width: 100%;
      min-width: 200px;
      > p {
        margin: 0 0 10px;
      }
    }
    .extra {
      padding: 0 20px 20px;
      text-align: left;
    }
    .statuses {
      padding: 10px 20px;
    }
    .toggle {
      display: flex;
      align-items: center;
      justify-content: center;
      flex: 0 0 30px;
      width: 30px;
      cursor: pointer;
      color: $black;
      transition: all .3s ease-in-out;
      background: #DEE0E3;
      &:hover {
        color: #fff;
        background: #A7A8AA;
        text-decoration: none;
      }
      .icon {
        margin: 0;
      }
    }
    h2 {
      margin: 0;
      font-size: 20px;
      line-height: 18px;
      font-weight: 900;
      word-break: break-all;
    }
    p {
      margin: 0;
    }
    a {
      color: $black;
      &:hover {
        text-decoration: underline;
      }
    }
    .data {
      display: flex;
      justify-content: center;
      flex-wrap: wrap;
      padding: 10px;
      background: #EAEBED;
      strong {
        display: block;
      }
      p {
        flex: 1 0 33.3333%;
        width: 33.3333%;
        min-width: 150px;
        padding: 0 10px 10px;
      }
    }
  }
  @include min-screen(768px) {
    &--wrap {
      padding: 0 10px;
    }
    &--item {
      width: 50%;
      padding: 0 10px;
      .link {
        padding: 40px 20px 20px;
      }
      h2 {
        font-size: 24px;
        line-height: 20px;
      }
    }
  }
  @include min-screen(1224px) {
    &--wrap {
      padding: 0 30px;
    }
    &--item {
      padding: 0 20px;
      margin-bottom: 40px;
    }
  }
}
.domains-list {
  position: relative;
  z-index: 2;
  display: flex;
  align-items: stretch;
  width: 1440px;
  max-width: 100%;
  .ui.table {
    tr {
      td {
        &:first-child {
          font-weight: 900;
          word-break: break-all;
          a {
            color: $black;
            &:hover {
              color: $base-blue;
            }
          }
        }
      }
    }
  }
  @include min-screen(768px) {
    margin: 0 auto 20px;
    padding: 0 20px;
  }
  @include min-screen(1224px) {
    margin: 0 auto 40px;
    padding: 0 50px;
  }
}