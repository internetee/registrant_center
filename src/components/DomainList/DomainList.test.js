import "core-js"
import React from "react"
import { mount, shallow } from "enzyme"
import DomainList from "./DomainList"
import contacts from "../../__mocks__/contacts"
import domains from "../../__mocks__/domains"
import Providers from "../../__mocks__/Providers"

describe("components/DomainList", () => {
    it("should render DomainsList", () => {
        const cmp = shallow(
            <Providers>
                <DomainList contacts={contacts} domains={domains} />
            </Providers>
        )
        expect(cmp).toMatchSnapshot()
    })

    it("should show domains grid on link click", () => {
        const cmp = mount(
            <Providers>
                <DomainList contacts={contacts} domains={domains} />
            </Providers>
        )
        cmp.find(".action--grid").at(0).simulate("click")
        expect(cmp).toMatchSnapshot()
        cmp.unmount()
    })

    it("should show domains list on link click", () => {
        const cmp = mount(
            <Providers>
                <DomainList contacts={contacts} domains={domains} />
            </Providers>
        )
        cmp.find(".action--list").at(0).simulate("click")
        expect(cmp).toMatchSnapshot()
        cmp.unmount()
    })

    it("should show domains filters on link click", () => {
        const cmp = mount(
            <Providers>
                <DomainList contacts={contacts} domains={domains} />
            </Providers>
        )
        cmp.find(".action--filter").at(0).simulate("click")
        expect(cmp).toMatchSnapshot()
        cmp.unmount()
    })

    it("should show domain extra info on link click", () => {
        const cmp = mount(
            <Providers>
                <DomainList contacts={contacts} domains={domains} />
            </Providers>
        )
        cmp.find(".toggle").at(0).simulate("click")
        expect(cmp).toMatchSnapshot()
        cmp.unmount()
    })
})
