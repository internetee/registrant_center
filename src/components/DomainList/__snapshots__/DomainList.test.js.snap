// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/DomainList should render DomainsList 1`] = `ShallowWrapper {}`;

exports[`components/DomainList should show domain extra info on link click 1`] = `ReactWrapper {}`;

exports[`components/DomainList should show domains filters on link click 1`] = `ReactWrapper {}`;

exports[`components/DomainList should show domains grid on link click 1`] = `ReactWrapper {}`;

exports[`components/DomainList should show domains list on link click 1`] = `ReactWrapper {}`;
